qiskit_experiments/curve_analysis/visualization/curves.py

```diff
@@ -61,36 +60,34 @@ def plot_curve_fit(
     if "linewidth" not in plot_opts:
         plot_opts["linewidth"] = 2
 
-    # Result data
-    fit_params = result.popt
-    param_keys = result.popt_keys
-    fit_errors = result.popt_err
     xmin, xmax = result.x_range
 
     # Plot fit data
     xs = np.linspace(xmin, xmax, num_fit_points)
-    if param_keys:
-        ys_fit = func(xs, **dict(zip(param_keys, fit_params)))
-    else:
-        ys_fit = func(xs, *fit_params)
-    ax.plot(xs, ys_fit, **plot_opts)
-
-    # Plot standard error interval
-    if fit_uncertainty and fit_errors is not None:
-        if param_keys:
-            params_upper = {}
-            params_lower = {}
-            for key, param, error in zip(param_keys, fit_params, fit_errors):
-                params_upper[key] = param + error
-                params_lower[key] = param - error
-            ys_upper = func(xs, **params_upper)
-            ys_lower = func(xs, **params_lower)
-        else:
-            params_upper = [param + error for param, error in zip(fit_params, fit_errors)]
-            params_lower = [param - error for param, error in zip(fit_params, fit_errors)]
-            ys_upper = func(xs, *params_upper)
-            ys_lower = func(xs, *params_lower)
-        ax.fill_between(xs, ys_lower, ys_upper, alpha=0.1, color=plot_opts["color"])
+    ys_fit_with_error = func(xs, **dict(zip(result.popt_keys, result.popt)))
+
+    # Line
+    ax.plot(xs, unp.nominal_values(ys_fit_with_error), **plot_opts)
+
+    # Confidence interval of 1 sigma
+    stdev_arr = unp.std_devs(ys_fit_with_error)
+    if np.isfinite(stdev_arr).all():
+        # 1 sigma plot
+        ax.fill_between(
+            xs,
+            y1=unp.nominal_values(ys_fit_with_error) - stdev_arr,
```

Comment:
I think these should still be an option so that you can choose the sigma levels to plot (or not plot any if you want). Eg if you want 1, 2, 3 you could put int `plot_stddevs = [1, 2, 3]` or some custom values like [0.5, 1.5] or whatever you like.